### PR TITLE
[DDW-159] Check for linting and flow-type errors in storybook directory

### PR DESCRIPTION
**Pull request by daniloprates**
This PR fixes linting and flow-type errors in the storybook directory.

## Todo:

- [x] Lint errors and warnings
- [x] Flow test errors

### Lint

* Before
image

* After
image

### Flow

* Before
image

* After
image

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`npm run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`npm run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`npm run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`npm run flow-test`)
- [ ] There are no *lint* errors or warnings (`npm run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `npm run manage-translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`npm run storybook`)
- [ ] In case of npm dependency changes both `package-lock.json` and `yarn.lock` files are updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by daniloprates** [comment]
@nikolaglumac Thanks for the help. The fixing is done. Please, let me know if there is anything that should be done differently.

File: flow/declarations/File.js [hunk 1]
```diff
@@ -2,3 +2,4 @@ declare class File {
   name: string,
   path: string,
 }
+export default File;
```
Comment (nikolaglumac) [inline flow/declarations/File.js]: @daniloprates why do we need this export here?

File: source/renderer/app/components/wallet/settings/export-to-file/WalletExportToFileDialog.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ export type OnSubmitParams = {
 type Props = {
   walletName: string,
   hasSpendingPassword: boolean,
-  isSubmitting: false,
+  isSubmitting: boolean,
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/settings/export-to-file/WalletExportToFileDialog.js]: @daniloprates great catch!

File: storybook/stories/AdaRedemptionDialog.stories.js [hunk 3]
```diff
@@ -28,21 +29,34 @@ storiesOf('AdaRedemptionForm', module)
...
         isCertificateSelected={false}
         isCertificateEncrypted={false}
         onCertificateSelected={action('onCertificateSelected')}
         onPassPhraseChanged={action('onPassPhraseChanged')}
         onRedemptionCodeChanged={action('onRedemptionCodeChanged')}
         onChooseRedemptionType={action('onChooseRedemptionType')}
-        redemptionCode=''
-        redemptionType='regular'
+        redemptionCode=""
+        redemptionType="regular"
```
Comment (nikolaglumac) [inline storybook/stories/AdaRedemptionDialog.stories.js]: @daniloprates please replace `regular` magic word with `ADA_REDEMPTION_TYPES.REGULAR` from this file: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/types/redemptionTypes.js#L5

File: storybook/stories/AdaRedemptionDialog.stories.js [hunk 4]
```diff
@@ -76,21 +103,34 @@ storiesOf('AdaRedemptionForm', module)
...
+        isCertificateSelected
+        isCertificateEncrypted
         onCertificateSelected={action('onCertificateSelected')}
         onPassPhraseChanged={action('onPassPhraseChanged')}
         onRedemptionCodeChanged={action('onRedemptionCodeChanged')}
         onChooseRedemptionType={action('onChooseRedemptionType')}
-        redemptionCode=''
-        redemptionType='regular'
+        redemptionCode=""
+        redemptionType="regular"
```
Comment (nikolaglumac) [inline storybook/stories/AdaRedemptionDialog.stories.js]: @daniloprates same as: https://github.com/input-output-hk/daedalus/pull/938/files#r190600805

File: storybook/stories/AdaRedemptionDialog.stories.js [hunk 5]
```diff
@@ -28,21 +29,34 @@ storiesOf('AdaRedemptionForm', module)
...
         getSelectedWallet={() => ({})}
         wallets={[
           { value: 'wallet-1', label: 'First Wallet' },
           { value: 'wallet-2', label: 'Second Wallet' },
           { value: 'wallet-3', label: 'Third Wallet' },
         ]}
+        suggestedMnemonics={[]}
```
Comment (nikolaglumac) [inline storybook/stories/AdaRedemptionDialog.stories.js]: @daniloprates please use `wordList` for suggestedMnemonics like here: https://github.com/input-output-hk/daedalus/blob/develop/storybook/stories/PaperWallets.stories.js#L59
Please do the same on other places as well. Thanks!

File: flow/declarations/File.js (hunk 1 above)
Comment (daniloprates) [reply flow/declarations/File.js]: It was returning a lint error. But couldn't reproduce, so removed in the latest commit.

File: source/renderer/app/components/wallet/settings/export-to-file/WalletExportToFileDialog.js (hunk 2 above)
Comment (daniloprates) [reply source/renderer/app/components/wallet/settings/export-to-file/WalletExportToFileDialog.js]: :)

File: storybook/stories/AdaRedemptionDialog.stories.js (hunk 3 above)
Comment (daniloprates) [reply storybook/stories/AdaRedemptionDialog.stories.js]: Done! ☝🏽

File: storybook/stories/AdaRedemptionDialog.stories.js [hunk 6]
```diff
@@ -52,21 +66,34 @@ storiesOf('AdaRedemptionForm', module)
...
+        isCertificateSelected
         isCertificateEncrypted={false}
         onCertificateSelected={action('onCertificateSelected')}
         onPassPhraseChanged={action('onPassPhraseChanged')}
         onRedemptionCodeChanged={action('onRedemptionCodeChanged')}
         onChooseRedemptionType={action('onChooseRedemptionType')}
-        redemptionCode=''
-        redemptionType='regular'
+        redemptionCode=""
+        redemptionType="regular"
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply storybook/stories/AdaRedemptionDialog.stories.js]: Done! ☝🏽

File: storybook/stories/AdaRedemptionDialog.stories.js (hunk 4 above)
Comment (daniloprates) [reply storybook/stories/AdaRedemptionDialog.stories.js]: Done! ☝🏽

File: storybook/stories/AdaRedemptionDialog.stories.js (hunk 5 above)
Comment (daniloprates) [reply storybook/stories/AdaRedemptionDialog.stories.js]: Done! ☝🏽

File: CHANGELOG.md [hunk 7]
```diff
@@ -15,6 +15,7 @@ Changelog
 
 ### Chores
 
+- Removed lint and flow errors and warnings ([PR 938](https://github.com/input-output-hk/daedalus/pull/938))
```
Comment (DominikGuzei) [inline CHANGELOG.md]: That description does not really reflect what the PR does right? It `added flow checks and linting for storybook code` from what i see?

**Comment by daniloprates** [comment]
@DominikGuzei that's completely right. That was what I first understood of the task. Done.

**Pull request merged**